api/v1alpha1/ztunnel_types.go

```diff
@@ -0,0 +1,198 @@
+// Copyright Istio Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package v1alpha1
+
+import (
+	"time"
+
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+
+const (
+	ZTunnelKind = "ZTunnel"
+)
+
+// ZTunnelSpec defines the desired state of ZTunnel
+type ZTunnelSpec struct {
+	// +sail:version
+	// Defines the version of Istio to install.
+	// Must be one of: v1.24.0 or latest.
+	// +operator-sdk:csv:customresourcedefinitions:type=spec,order=1,displayName="Istio Version",xDescriptors={"urn:alm:descriptor:com.tectonic.ui:fieldGroup:General", "urn:alm:descriptor:com.tectonic.ui:select:v1.24.0", "urn:alm:descriptor:com.tectonic.ui:select:latest"}
+	// +kubebuilder:validation:Enum=v1.24.0;latest
+	// +kubebuilder:default=v1.24.0
+	Version string `json:"version"`
+
+	// +sail:profile
+	// The built-in installation configuration profile to use.
+	// The 'default' profile is always applied. On OpenShift, the 'openshift' profile is also applied on top of 'default'.
+	// Must be one of: ambient, default, demo, empty, external, openshift-ambient, openshift, preview, remote, stable.
+	// +++PROFILES-DROPDOWN-HIDDEN-UNTIL-WE-FULLY-IMPLEMENT-THEM+++operator-sdk:csv:customresourcedefinitions:type=spec,displayName="Profile",xDescriptors={"urn:alm:descriptor:com.tectonic.ui:fieldGroup:General", "urn:alm:descriptor:com.tectonic.ui:select:ambient", "urn:alm:descriptor:com.tectonic.ui:select:default", "urn:alm:descriptor:com.tectonic.ui:select:demo", "urn:alm:descriptor:com.tectonic.ui:select:empty", "urn:alm:descriptor:com.tectonic.ui:select:external", "urn:alm:descriptor:com.tectonic.ui:select:minimal", "urn:alm:descriptor:com.tectonic.ui:select:preview", "urn:alm:descriptor:com.tectonic.ui:select:remote"}
+	// +operator-sdk:csv:customresourcedefinitions:type=spec,xDescriptors={"urn:alm:descriptor:com.tectonic.ui:hidden"}
+	// +kubebuilder:validation:Enum=ambient;default;demo;empty;external;openshift-ambient;openshift;preview;remote;stable
+	Profile string `json:"profile,omitempty"`
+
+	// Namespace to which the Istio ztunnel component should be installed.
+	// +operator-sdk:csv:customresourcedefinitions:type=spec,xDescriptors={"urn:alm:descriptor:io.kubernetes:Namespace"}
+	// +kubebuilder:default=kube-system
```

Comment:
Is this correct? Do we want to deploy ztunnel to `kube-system`? For `IstioCNI` we use `istio-cni` instead of `kube-system`.

Reply:
If we change the namespace to `istio-cni`, we will also have to modify the trustedZtunnelNamespace which is configured to `kube-system`.

Reply:
I don't think it's the best idea to use `kube-system` on OpenShift, as all the other OpenShift components intentionally typically use other namespaces, never `kube-system`.

Reply:
Agree with Marko. In that case, we should probably update the upstream charts/docs as well to use `istio-cni` namespace instead of `kube-system`. 

@yxun when you tried installing ztunnel in `istio-cni`, did you also configure `trustedZtunnelNamespace` to `istio-cni`? If so, did it work or did you notice any additional errors?

Reply:
> I don't think it's the best idea to use `kube-system` on OpenShift, as all the other OpenShift components intentionally use other namespaces, never `kube-system`.

@luksa, I see that when using istioctl with `openshift` profile (i.e., non ambient use-case) the `istio-cni` pods are installed in the `kube-system` namespace itself and not in `istio-cni` namespace.

Reply:
I updated this default namespace value in the `ztunnel_types.go` file.
The Helm chart profile values need to be updated from upstream if we want all openshift related profiles use `trustedZtunnelNamespace: istio-cni`

Reply:
As discussed during the call yesterday, we will be installing the ztunnel CR in the `ztunnel` namespace and not in `istio-cni`. I will take care of trustedZtunnelNamespace validation/modification as part of my PR.